staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -0,0 +1,303 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kuberc
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+	"k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/config/v1alpha1"
+	"k8s.io/kubectl/pkg/scheme"
+)
+
+const RecommendedKubeRCFileName = ".kuberc"
+
+var (
+	RecommendedConfigDir  = filepath.Join(homedir.HomeDir(), clientcmd.RecommendedHomeDir)
+	RecommendedKubeRCFile = filepath.Join(RecommendedConfigDir, RecommendedKubeRCFileName)
```

Comment:
This gives you `$HOME/.kube/.kuberc`, I'd propose either `$HOME/.kube/kuberc` or `$HOME/.kuberc`. Given we want to isolate kubeconfig from kuberc I'd probably lean towards the latter.

Reply:
Updated to `$HOME/.kube/kuberc`, I can change it to something different if we decide to another format.

Reply:
should we consider https://github.com/kubernetes/kubernetes/issues/56402 if we're starting a new file?

Reply:
I'm fine to change the default path to `~/.config/kube/kuberc`. In my opinion, people get used to look at under `~/.kube` currently and until we change the default for everything, it would be better to store kuberc under `~/.kube/` as well.

Reply:
> should we consider #56402 if we're starting a new file?

We've had a long discussion about the placement of the file. And like said, the majority of folks is just used to the current location and that's how we ended up with it.

Reply:
ok, as long as we took the opportunity to consider it

Reply:
@liggitt would you be ok to tag the openapi changes in that case?. I think, this is the only bit that hasn't been approved yet. Thanks.